File: Python/2367.py
# 2367. Number of Arithmetic Triplets
# https://leetcode.com/problems/number-of-arithmetic-triplets/

class Solution:
  def arithmeticTriplets(self, nums: List[int], diff: int) -> int:
    length = len(nums)
    out = 0

    for i in range(0, length - 2):
      for j in range(i + 1, length - 1):
        if (nums[j] - nums[i] < diff):
          continue
        elif (nums[j] - nums[i] > diff):
          break
        else:
          for k in range(j + 1, length):
            if (nums[k] - nums[j] < diff):
              continue
            elif (nums[k] - nums[j] > diff):
              break
            else:
              out += 1
              break

    return out

File: TypeScript/2367.ts
/**
 * 2367. Number of Arithmetic Triplets
 * https://leetcode.com/problems/number-of-arithmetic-triplets/
 */

function arithmeticTriplets(nums: number[], diff: number): number {
  let len: number = nums.length;
  let out: number = 0;

  for (let i: number = 0; i < len - 2; i++) {
    for (let j: number = i + 1; j < len - 1; j++) {
      if (nums[j] - nums[i] < diff) {
        continue;
      } else if (nums[j] - nums[i] > diff) {
        break;
      } else {
        for (let k: number = j + 1; k < len; k++) {
          if (nums[k] - nums[j] < diff) {
            continue;
          } else if (nums[k] - nums[j] > diff) {
            break;
          } else {
            out += 1;
            break;
          }
        }
      }
    }
  }

  return out;
}
